gnovm/pkg/gnolang/op_expressions.go

```diff
@@ -680,7 +657,7 @@ func (m *Machine) doOpFuncLit() {
 	// every invocation of the function.
 	captures := []TypedValue(nil)
 	for _, nx := range x.HeapCaptures {
-		ptr := lb.GetPointerTo(m.Store, nx.Path)
+		ptr := lb.GetPointerToDirect(m.Store, nx.Path)
 		// check that ptr.TV is a heap item value.
 		// it must be in the form of:
 		// {T:heapItemType{},V:HeapItemValue{...}}
```

Comment:
this panics, it looks like the Nx define in IfStmt.Init is not set heapDefine attr.

```go
package main

func trymake() func() {
	if s := "hello"; true {
		return func() {
			println(s)
		}
	}
	return func() {}
}

func main() {
	trymake()()
	println("---ok")
}

```

Reply:
this causes stdlib/bytes test failed.

Reply:
trying to do a fix here:
https://github.com/gnolang/gno/pull/4095/commits/495a2f411b8baa79f14bf5420c5864da1f9ea31b